src/initialization/InitAmrCore.cpp

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2022 The Regents of the University of California, through Lawrence
+ *           Berkeley National Laboratory (subject to receipt of any required
+ *           approvals from the U.S. Dept. of Energy). All rights reserved.
+ *
+ * This file is part of ImpactX.
+ *
+ * Authors: Axel Huebl, Chad Mitchell, Ji Qiang
+ * License: BSD-3-Clause-LBNL
+ */
+#include "InitAmrCore.H"
+
+#include "initialization/InitAMReX.H"
+
+#include <AMReX_Array.H>
+#include <AMReX_Box.H>
+#include <AMReX_CoordSys.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_ParallelDescriptor.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_RealBox.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SPACE.H>
+#include <AMReX_Vector.H>
+
+#include <stdexcept>
+
+
+namespace impactx::initialization
+{
+namespace details
+{
+
+    /** Set the initial physical simulation extent
+     *
+     * Sets to either a fake min:max box or to the user-provided extent.
+     * Controlled by geometry.dynamic_size.
+     *
+     * @return the RealBox in meters
+     */
+    amrex::RealBox
+    init_physical_domain ()
+    {
+        amrex::ParmParse pp_geometry("geometry");
+        bool dynamic_size = true;
+        pp_geometry.queryAdd("dynamic_size", dynamic_size);
+        amrex::Vector<amrex::Real> prob_lo(undefined_geometry_prob_lo.begin(), undefined_geometry_prob_lo.end());
+        amrex::Vector<amrex::Real> prob_hi(undefined_geometry_prob_hi.begin(), undefined_geometry_prob_hi.end());
+        if (dynamic_size)
+        {
+            pp_geometry.queryAdd("prob_lo", prob_lo);
```

Comment:
This could be a bit misleading, no?
It is seems that, in the `dynamic_size` mode, it is reading the `prob_lo` and `prob_hi`, but this actually has no influence on the simulation, because `prob_lo` and `prob_hi` will be redefined anyway by `ResizeMesh`.

Reply:
I mean to add a default here in case someone reads the value before `evolve()` is started (I think some internal routines for particle init do). Maybe `addarr` is less confusing.

Reply:
```suggestion
            // read and conditionally overwrite problem domain size:
            //   users might switch to !dynamic_size after init and we don't want to overwrite
            //   their inputs
            pp_geometry.queryAdd("prob_lo", prob_lo);
```